### DISCUSSION TITLE
From Ugs to bCNC homing limit switches problem

### DISCUSSION THREAD

**Discussion by atomino-max** [vlachoudis/bCNC#1634] [General]
I am running grbl 1.1g and my cnc is working fine with Ugs.I should want use bCNC and moved the firmware setting from Ugs to bCNC, but during X home position the system cannot detect the limit switch position.
Any idea ?

![ugs](https://user-images.githubusercontent.com/79934655/137535990-0870d815-1712-4848-b70a-08ca04d312b9.jpg)
![bcnc](https://user-images.githubusercontent.com/79934655/137537228-cea7b3b8-23f3-4d4b-87bb-917166bb4c9b.jpg)

**Comment by JipeB** [comment]
Hello,

bCNC only reads the information transmitted by Grbl. If there is no feedback on X, I think we should start by checking the continuity of the limit switch circuit of this axis.

Pressing the switch manually causes a message to appear in the CNC status bar (Control tab).

JP

**Comment by atomino-max** [comment]
If I run Ugs the system can make homing without any problem.
May be the problem is the hardware configuration, I am running Grbl with Mega 2560 and ramps 1.4.
I attach the file modified by me on Grbl, can be possible to run bCnc ?
[config.h.txt](https://github.com/vlachoudis/bCNC/files/7367418/config.h.txt)
[cpu_map.h.txt](https://github.com/vlachoudis/bCNC/files/7367420/cpu_map.h.txt)

**Comment by JipeB** [comment]
I am using arduino uno and I am not familiar with your system. 
I noticed that your configuration enabled Hard Limits; does your machine have 6 EndStops?
If so, the parameters $130, $131, and $132 are not needed.
If your machine has only 3 endStops, you must deactivate Hard Limits and validate Soft Limits, the parameters $130, $131 and $132 defining the useful strokes of each axis.
I saw on some forums that the operation of EndStops was problematic on the mega2560 cards associated with Ramps; maybe a line of research ...

**Comment by atomino-max** [comment]
I have 6 limit switches
I have changed $10 Status report [mask] to 19 and now with command ? I have in the Terminal but not in the CNC status bar, the state of each limit switches  for home position Pn:XYZ  instead no any message for limit switches in max position, see below.
So it means that the signals are detected from grbl but not from bCnc.
Do you have any suggestions ?

?
ok
b'$G\n'
[GC:G0 G54 G17 G21 G90 G94 M5 M9 T0 F0 S0]
ok
<Idle|MPos:0.000,0.000,0.000|Bf:35,254|FS:0,0|Pn:X|Ov:100,100,100>
?
ok
b'$G\n'
[GC:G0 G54 G17 G21 G90 G94 M5 M9 T0 F0 S0]
ok
<Idle|MPos:0.000,0.000,0.000|Bf:35,254|FS:0,0|Pn:XZ>
?
ok
b'$G\n'
[GC:G0 G54 G17 G21 G90 G94 M5 M9 T0 F0 S0]
ok
<Idle|MPos:0.000,0.000,0.000|Bf:35,253|FS:0,0|Pn:XYZ>
?

**Reply by nealre** [reply]
Hi, as I read all these problems it touches mine as well. I can't get the gantry to goto the home position and strike the limit switch on the "Y" axes and stop. It tries to keep going even though the switch activated the alarm it still didn't stop.  Also, on the "X" axis when it hits it doesn't bounce back enough and causes the machine to loose it's connection and I have to reset the USB connection again. I believe the main problem is in the firmware. I only need the X and Y for my 2.5 watt Laser. I have set that up ok in the "config.h" file. The problem is, I believe ,there are other settings that need set as well. The Arduino nano is near the limit of it's memory. help!

**Comment by JipeB** [comment]
$ 10 = 19 is the default setting of grbl (I use it on my machine)

On the other hand, the parameter $ 2 = 0 calls out to me, by default it is at 25 ms; this is the time the stepper motors are kept energized after each movement command. To avoid modifying the position of my axes when changing tools $ 2 = 255 on my machine (motors permanently powered on, therefore locked)
You could try resetting this parameter to 25.

Default settings for:
$ 11 = 0.010 mm
$ 12 = 0.002 mm (your setup value is huge)

I have no other leads at the moment. This is what I get on my machine at the end of the homing cycle.

**Comment by JipeB** [comment]
Sorry, I cannot post the screenshot.

**Comment by JipeB** [comment]
With this computer it is better for the images
![POM](https://user-images.githubusercontent.com/91841136/138086411-e24034fc-3864-49e9-971a-6f53d07b9112.jpg)

**Comment by atomino-max** [comment]
I fixed the problem, it was my fault, I pushed XY Origin button instead of Home button.
Thanks a lot for your support.
Have a nice day
Max

**Comment by JipeB** [comment]
The discovery of new software is often subject to a few uncertainties ...
bCNC is an interesting tool, but unfortunately little or poorly documented. For those who can, reading the code makes it possible to understand the author's approach and to remove certain uncertainties as to the implementation.
Have a good day and look forward to it.
JP
